lib/webdriver.py

```diff
@@ -260,6 +314,50 @@ def _get_needed_headers(self, request_headers: JSON) -> JSON:
 
         return headers
 
+    def _is_cache_expired(self, cached_time_str: str, current_time: datetime) -> bool:
+        try:
+            cached_time = datetime.fromisoformat(cached_time_str)
+            return current_time - cached_time > self.cache_duration
+        except ValueError:
+            logger.error(f"Invalid timestamp format: {cached_time_str}")
+            return True
+
+    def _cached_headers_manager(
+        self,
+        current_time: Optional[datetime] = None,
+        login_failed: Optional[bool] = None,
+        headers: Optional[JSON] = None,
+    ) -> JSON:
+        """
+        Read existing cached data from JSON file or update it with new values.
+        """
+        if os.path.exists(self.cached_headers_path):
+            try:
+                with open(self.cached_headers_path, "r") as file:
+                    cached_data = json.load(file)
+            except json.JSONDecodeError:
```

Comment:
Same as my comment below about the IOError. I think we should fail here as this file and its contents are controlled fully within the script. Therefore, it should either not reach a decode error or it should crash because there's a bug in the program (again, I'd rather have it crash than log an error as per the below comment too).